gear/gear/cloud_config.py

```diff
@@ -3,6 +3,20 @@
 import json
 
 
+class AzureConfig:
```

Comment:
Is it possible to unify this with AzureInstanceEnvironment? I'd prefer not to have lots of subtly different classes if we can help it.

Reply:
I intended these config objects mainly as a way of getting python types on the k8s secret for each cloud. I agree that it is better to have a robust abstraction that can provide functionality, like AzureInstanceEnvironment was turned into, rather than passing a dict around everywhere informing different components how to behave. But after your earlier comments, we've transformed AzureInstanceEnvironment to the AzureWorkerAPI, which seems like it shouldn't be used more widely than the worker.

If we had something else like a AzureDriverAPI, and both API abstractions used the AzureConfig just to initialize themselves, would that be reasonable?

Reply:
Overall I think the cloud configs had a bit of a tangled history and I have yet to be convinced of their ratio of utility : danger. I think as long as the pieces of configuration (resource_group, etc.) are hidden behind powerful abstractions like the AzureWorkerAPI or AzureDriver, whether they get them this way or just through env variables is not as important.

Reply:
Unless you see a clearly better solution, I'm happy to do a separate pass once this is all in to see what's actually useful here.